### ISSUE TITLE
Error trying to get a JWT token from Dockerhub for the stored Docker login.

### ISSUE THREAD

**Issue by boredazfcuk** [christian-korneck/docker-pushrm#6]
When attempting to push the README.md file to Dockerhub, I get the following error:
```
ERRO error trying to get a JWT token from Dockerhub for the stored Docker login. Try "docker logout" and "docker login". Also, if you have 2FA auth enabled in Dockerhub you'll need to disable it for this tool to work. (This is an unfortunate Dockerhub limitation, see docs for more infos).
```
Docker-pushrm version:
```
pushrm*     Push Readme to container registry (Christian Korneck, 1.4.2)
```
Docker CLI version and platform:
```
Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:45:50 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:44:21 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
Exact command that you're running:
```
docker pushrm boredazfcuk/icloudpd
```
Debug output:
```
DEBU root cmd init config
DEBU home dir: /root
DEBU subcommand "pushrm" called
DEBU Using target: docker.io/boredazfcuk/icloudpd:latest
DEBU using README file: README.md
DEBU server: docker.io
DEBU namespace: boredazfcuk
DEBU repo: icloudpd
DEBU tag: latest
DEBU repo provider: dockerhub
DEBU Dockerhub.GetAuthident called
DEBU no credentials found in env vars. Trying Docker credentials store
DEBU Using config file: /root/.docker/config.json
DEBU util.GetDockerCreds called
DEBU tried candidate https://index.docker.io/v1/: found credentials for user boredazfcuk
DEBU Using Docker creds: boredazfcuk ********
DEBU Dockerhub.Pushrm called
DEBU retrieve Dockerhub jwt token, status code: 401
DEBU error retrieving Dockerhub jwt token, bad status code for response: 401 Unauthorized
ERRO error trying to get a JWT token from Dockerhub for the stored Docker login. Try "docker logout" and "docker login". Also, if you have 2FA auth enabled in Dockerhub you'll need to disable it for this tool to work. (This is an unfortunate Dockerhub limitation, see docs for more infos).
```

Additional context:
2FA is not enabled.
I'm building multi-arch images using buildx which successfully pushes to Dockerhub using same stored password.
`docker logout` and `docker login` has been tried but results are the same.

**Comment by christian-korneck** [comment]
hi @boredazfcuk , thanks for reaching out.

I just tried to reproduce your problem with multiple Docker accounts, but it worked fine for me.

First of all, are you sure that you're using the password for your Docker account and not a personal access token? (I'm just asking because it's possible to create personal access tokens even if you don't have 2FA auth enabled).

There are two more things you could try:

- just in case there's a corruption of the Docker config file (where to login is stored), delete/rename it and re-login (which recreates the file):
```
mv /root/.docker/config.json /root/.docker/config.json.backup
docker login
```
(I don't really expect it will help as in your logs it looks like the username could get extracted just fine, but it's worth to try)

- to be sure that there's no weird encoding bug regarding the password, could you maybe try to (temporarily) change your Docker password to a simple, alphanumeric one (without any special characters)?

(And if it works with a simple password it would be great if you could tell me the special characters from your non-working password so that I can try to reproduce the problem. If there's a bug I'll fix it).

**Comment by boredazfcuk** [comment]
Hi, Christian.

Thanks for taking time to look at this for me.

I'm pretty sure it's not a personal access token, as the Dockerhub security page shows:
```
It looks like you have not created any access tokens.
Docker Hub lets you create tokens to authenticate access. Treat personal access tokens as alternatives to your password. Learn more
```

I'm actually rebuilding my entire container collection with `docker buildx`, at the minute (making them all multi-arch, instead of amd64 only) so if I change my password it will probably kill that process off. It's been going for nearly two hours now so I'll wait for that to complete before I try removing and recreating the config.json file. It's contents look fine though:

```
{
        "auths": {
                "https://index.docker.io/v1/": {
                        "auth": "Rand0MchaRACT3RsQ2Fib3Q="
                }
        },
        "HttpHeaders": {
                "User-Agent": "Docker-Client/19.03.12 (linux)"
        }
}
```

I do have a special character in my password. Once my builds are finished, I'll try removing it and seeing what happens.

**Comment by christian-korneck** [comment]
> I do have a special character in my password

interesting. May I ask what character exactly?

**Comment by boredazfcuk** [comment]
It’s a :

**Comment by christian-korneck** [comment]
thanks, I could now reproduce it. This should now be fixed in [v1.4.3](https://github.com/christian-korneck/docker-pushrm/releases/tag/v1.4.3). Can you try and confirm if this resolves your problem?

**Comment by boredazfcuk** [comment]
Yep, it's all good now. I've just checked Dockerhub and my new readme has landed.

Thanks for sorting.
